File: BW8AGU/VB/Ch08/BW8AGU.Samples.VB/BW8AGU.Samples.VB/Common/VersionHelper.vb
Public NotInheritable Class VersionHelper
    Private Sub New()
    End Sub
    Public Shared Sub CheckVersion()
        Dim version As UInteger = ApplicationData.Current.Version

        Select Case version
            Case 0
                'Need to upgrade data from v0 to v1
                Upgrade_Version0_to_Version1()
                Exit Select
            Case 1
                'Right version, do nothing...
                Exit Select
            Case Else
                Throw New Exception("Unexpected ApplicationData Version: " + version)
        End Select
    End Sub

    Private Shared Async Sub Upgrade_Version0_to_Version1()
        Await ApplicationData.Current.SetVersionAsync(1,
                                                      New ApplicationDataSetVersionHandler(AddressOf SetVersion1Handler))
    End Sub

    Private Shared Sub SetVersion1Handler(setVersionRequest As SetVersionRequest)
        Dim deferral As SetVersionDeferral = setVersionRequest.GetDeferral()

        'Change the data format for all needed settings
        ApplicationData.Current.LocalSettings.Values("LocalData") = "this-is-a-new-data-format"

        deferral.Complete()
    End Sub

    Public Shared ReadOnly Property AppVersion() As UInteger
        Get
            Return ApplicationData.Current.Version
        End Get
    End Property
End Class

File: BW8AGU/C#/Ch08/BW8AGU.Samples.CSharp/Common/VersionHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Storage;

namespace BW8AGU.Samples.CSharp.Common
{
    public static class VersionHelper
    {
        public static void CheckVersion()
        {
            uint version = ApplicationData.Current.Version;

            switch (version)
            {
                case 0:
                    //Need to upgrade data from v0 to v1
                    Upgrade_Version0_to_Version1();
                    break;
                case 1:
                    //Right version, do nothing...
                    break;
                default:
                    throw new Exception("Unexpected ApplicationData Version: " + version);
            }
        }

        static async void Upgrade_Version0_to_Version1()
        {
            await ApplicationData.Current.SetVersionAsync(1,
                new ApplicationDataSetVersionHandler(SetVersion1Handler));
        }

        private static void SetVersion1Handler(SetVersionRequest setVersionRequest)
        {
            SetVersionDeferral deferral = setVersionRequest.GetDeferral();

            //Change the data format for all needed settings
            ApplicationData.Current.LocalSettings.Values["LocalData"] = "this-is-a-new-data-format";

            deferral.Complete();
        }

        public static uint AppVersion
        {
            get { return ApplicationData.Current.Version; }
        }

    }
}
